base_layer/core/src/validation/test.rs

```diff
@@ -422,16 +422,33 @@ fn chain_balance_validation_burned() {
 
 mod transaction_validator {
     use super::*;
+    use crate::transactions::transaction_components::TransactionError;
 
     #[test]
     fn it_rejects_coinbase_outputs() {
         let consensus_manager = ConsensusManagerBuilder::new(Network::LocalNet).build();
         let db = create_store_with_consensus(consensus_manager);
         let factories = CryptoFactories::default();
         let validator = TxInternalConsistencyValidator::new(factories, true, db);
-        let features = OutputFeatures::create_coinbase(0);
+        let features = OutputFeatures::create_coinbase(0, None);
         let (tx, _, _) = tx!(MicroTari(100_000), fee: MicroTari(5), inputs: 1, outputs: 1, features: features);
         let err = validator.validate(&tx).unwrap_err();
         unpack_enum!(ValidationError::ErroneousCoinbaseOutput = err);
     }
+
+    #[test]
+    fn coinbase_extra_must_be_empty() {
+        let consensus_manager = ConsensusManagerBuilder::new(Network::LocalNet).build();
+        let db = create_store_with_consensus(consensus_manager);
+        let factories = CryptoFactories::default();
+        let validator = TxInternalConsistencyValidator::new(factories, true, db);
+        let mut features = OutputFeatures { ..Default::default() };
+        features.coinbase_extra = b"deadbeef".to_vec();
+        let (tx, _, _) = tx!(MicroTari(100_000), fee: MicroTari(5), inputs: 1, outputs: 1, features: features);
+        let err = validator.validate(&tx).unwrap_err();
+        assert!(matches!(
+            err,
+            ValidationError::TransactionError(TransactionError::NonCoinbaseHasOutputFeaturesCoinbaseExtra)
+        ));
```

Comment:
This test was failing before adding the change to base_layer/core/src/validation/transaction_validators.rs#76